js/ui/appDisplay.js

```diff
@@ -1117,10 +1117,13 @@ class AppDisplay extends BaseAppView {
             let icon = this._items.get(id);
             if (!icon) {
                 icon = new FolderIcon(id, path, this);
-                icon.connect('apps-changed', () => {
+                const folderAppsChangedId = icon.connect('apps-changed', () => {
```

Comment:
Why? The signal is disconnected when the source object is disposed.

Reply:
But iirc we don't destroy the icons explicitly when destroying the App display?

Reply:
I am certain I did hit this issue a few times while testing the drag&drop crash but haven't tested after the gjs update.

Reply:
Clutter recursively destroys actors. In the one case the folder icon isn't added to the grid, it is immediately destroyed. There's just no way for a folder icon to outlive the icon grid. It shouldn't happen with older GJS, it shouldn't happen with newer GJS either.

Reply:
Ok, I can drop this change but I'm 100% sure I hit this while testing and not only once but multiple times.

Reply:
Dropped this one and we can re-add at a later point if we hit it again - it could be because of the old gjs playing tricks with the GC/backtrace :P.